vertical-pod-autoscaler/recommender/model/cluster.go

```diff
@@ -0,0 +1,98 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package model
+
+import (
+	labels "k8s.io/apimachinery/pkg/labels"
+)
+
+// Cluster holds all runtime information about resources in the cluster
+// required for VPA operations.
+type Cluster struct {
+	// Pods in the cluster.
+	Pods map[PodID]*Pod
+}
+
+// Pod holds runtime information about a single Pod.
+type Pod struct {
+	// Unique id of the Pod.
+	ID PodID
+	// Set of labels attached to the Pod.
+	Labels labels.Set
+	// Containers that belong to the Pod, keyed by the container name.
+	Containers map[string]*ContainerStats
+}
+
+// NewCluster returns a new Cluster with no pods.
+func NewCluster() *Cluster {
+	return &Cluster{
+		make(map[PodID]*Pod), // empty pods map.
+	}
+}
+
+// ContainerUsageSampleWithKey holds a ContainerUsageSample together with the
+// ID of the container it belongs to.
+type ContainerUsageSampleWithKey struct {
+	ContainerUsageSample
```

Comment:
Please resolve the naming with @KarolKraskiewicz. I would like to aviod having multiple structs with names like:
* ContainerUsageSnapshot
* ContainerUtilisationSnapshot
* ContainerUsageSample

Reply:
I renamed `containerUsageSnapshot` to `containerMetricsSnapshot` but it is just an internal object. 
The only one which is exposed is `ContainerUtilisationSnapshot` which is somehow similar to `ContainerUsageSample` but `ContainerUtilisationSnapshot` contains some more information like:
* duration of the measurement
* docker image name
* pod creation time
* requested resources (in addition to actual usage)

`ContainerUtilisationSnapshot` is also different in a way that it aggregates data from pods and containers into single, flat structure while here we have some hierarchy. I could refactor Metrics client in a way it produces more hierarchical model if I had a better understanding how is it going to be consumed.

Reply:
Done. Our code now shares the same structures, defined in types.go.